Sources/Networking/HTTPClient/ETagManager.swift

```diff
@@ -86,7 +86,8 @@ class ETagManager {
                 let newResponse = storedResponse.withUpdatedValidationTime()
 
                 self.storeIfPossible(newResponse, for: request)
-                return newResponse.asResponse(withRequestDate: response.requestDate)
+                return newResponse.asResponse(withRequestDate: response.requestDate,
+                                              headers: response.responseHeaders)
```

Comment:
This will use the cached response but with the 304 response headers... I think this could be confusing... 

I was thinking, we could maybe add additional properties to `HTTPResponse` to have the original headers and new headers (only on 304 responses). It would still be a bit confusing, but clearer that there are 2 sets of headers.

Another option would be to rename the property in HTTPResponse to `latestHeaders` or something and add some documentation... Thoughts?

Reply:
> This will use the cached response but with the 304 response headers... I think this could be confusing...

I mean this method is `httpResultFromCacheOrBackend`, it returns the cached value if getting a 304.
Another way to think about it is, this method returns the 304 response + the "missing body".

I'll add a comment mentioning that.

Reply:
Done: 25510ab
And also updated a test to reflect this when using the mock: d6ac1be

Reply:
Yeah I think that helps 👍